Add mockServices docs (#26295)
backstage/backstage: docs/backend-system/building-plugins-and-modules/02-testing.md

bound listening port. You can use this to perform lower level network
interactions with the running test service.

### mock services

> The `mockServices` object provides a number of service factory functions satisfy plugin dependencies but also mocks to verify interactions between plugin and services.

Reviewer: ```suggestion
The `mockServices` object from `@backstage/backend-test-utils` provides a number of service factory functions that you can use to verify interactions between plugin and services.
```

> The following mock services are available:

Reviewer: Might be worth just saying that ALL core services have a corresponding mock? If you have the list, could they link to eg `https://backstage.io/docs/reference/backend-test-utils.mockservices.auth` etc? Those are very sparse now but at least there's something that we could add more doc comments to later down the line.

Could also show that basically all of them have a `factory` and a `mock`, and how those two differ from each other. Using a `mock({ someMethod: () => value })` or whatnot could be cool too just to display that you can send in partial implementations to it.

Author: Great ideas!
